database-commons/src/main/java/co/cask/ConnectionConfig.java

```diff
@@ -116,15 +103,23 @@ public static Properties getConnectionArguments(@Nullable String connectionArgum
       connectionArgumentsMap.put("user", user);
       connectionArgumentsMap.put("password", password);
     }
+
     Properties properties = new Properties();
     properties.putAll(connectionArgumentsMap);
+
     return properties;
   }
 
   /**
    * @return a {@link Properties} of connection arguments, parsed from the config.
    */
   public Properties getConnectionArguments() {
-    return getConnectionArguments(connectionArguments, user, password);
+    Properties arguments = getConnectionArguments(this.connectionArguments, user, password);
+    arguments.putAll(getDBSpecificArguments());
+    return arguments;
   }
+
+  public Map<String, String> getDBSpecificArguments() {
```

Comment:
It's now called by other classes. Since we can't directly put host, name and port into ConnectionConfig (otherwise these fields will be displayed for generic DB plugin) we now provide them as DB-specific via Map.

Reply:
Why is this called and not getConnectionProperties()? I thought this was meant to be implemented by the DB specific configs to translate their DB specific properties into generic connection properties.